galore/cross_sections.py

```diff
@@ -281,3 +303,242 @@ def _eval_fit(energy, coeffs):
+         
+                ## get highest cross section of this obital
+                highest_obital_cross_section = np.max(np.nan_to_num(result))
+                orbitals_cross_sections_dict[orbital] = highest_obital_cross_section
+                
+            elif dataset == 'Yeh':
+                obital_cross_sections = interm_matrix[energy_index]
+                ## get unit cross sections of this orbital
+                unit_cross_sections = obital_cross_sections/electrons_number
+                ## get highest cross section of this obital
+                highest_obital_cross_section = np.max(np.nan_to_num(unit_cross_sections))
+                orbitals_cross_sections_dict[orbital] = highest_obital_cross_section
+            
+                
+    return orbitals_cross_sections_dict
+
+def _get_metadata(energy,dataset):
+    """
+    Args:
+        energy(float): energy value  
+        dataset(str): 'Scofield' or 'Yeh'
+        
+        Note: 1.'Scofield' for J. H. Scofield (1973)
+                Lawrence Livermore National Laboratory Report No. UCRL-51326              
+              2.'Yeh' for Yeh, J.J. and Lindau, I. (1985) 
+                Atomic Data and Nuclear Data Tables 32 pp 1-155   
+                
+    Returns:
+        metadata_dict: containing the input energy value 
+                       and description of input reference
+  
+    """   
+    
+    metadata_dict = {}
+    metadata_dict['energy'] = energy
+    if dataset == 'Scofield':
+        metadata_dict['citation'] = 'J.H. Scofield, Theoretical photoionization cross sections from 1 to 1500 keV'
+        metadata_dict['link'] = 'https://doi.org/10.2172/4545040' 
+    elif dataset == 'Yeh':
+        metadata_dict['citation'] = 'Yeh, J.J. and Lindau, I. (1985) Atomic Data and Nuclear Data Tables 32 pp 1-155'
+        metadata_dict['link'] = 'https://doi.org/10.1016/0092-640X(85)90016-6'
+    else:
+        metadata_dict('dataset error: you can enter reference as "Scofield" or "Yeh" ' )
+    return metadata_dict
+
+
+
+def galore_install_data(DOI,data_file_dir,data_file_path):
+    
+    
+    if os.path.isfile(data_file_path)== True:
+        print("Data file exists.")
+        
+    else:
+        print("Downloading file...")
+        
+        
+        try:
+            os.mkdir(data_file_dir)
+        except:
+            pass
+        
+        urllib.request.urlretrieve(DOI,data_file_path)
+        
+    
+        if os.path.isfile(data_file_path) == True:
+            print("Done")
+            
+
+def get_csv_file_path(dataset):
+    
+    if platform.system()== "Windows":
+        
+        data_file_dir = os.path.join(os.getenv('LOCALAPPDATA'),'galore_data')
+    else:
+        data_file_dir = os.path.join(os.path.expanduser('~'),'.galore_data')
+        
+    if dataset == 'Scofield':
+        data_file_path = os.path.join(data_file_dir,'Scofield_csv_database.zip')
+        DOI = 'https://ndownloader.figshare.com/articles/15081888/versions/1'
```

Comment:
It's a bit misleading to refer to this as `DOI` because it is actually a direct link. `url` would be clearer; `DOI` implies that additional logic will be needed to find the correct URL.